### PR TITLE
transform and trainer namespaces don't need to follow the catalogs

### PR DESCRIPTION
**Pull request by sfilipi**
Microsoft.Ml.Transforms.Normalizers
Microsoft.Ml.Transforms.Categoricals
Microsoft.Ml.Transforms.Conversions
Microsoft.Ml.Transforms.Projections

into Microsoft.Ml.Transforms

Microsoft.ML.Trainers.KMeans
Microsoft.ML.Trainers.PCA
Microsoft.ML.Trainers.OnlineLearners
Microsoft.ML.Trainers.FactorizationMachine

into Microsoft.ML.Trainers

Addresses #2751.

### CODE REVIEW

File: src/Microsoft.ML.FastTree/GamModelParameters.cs [hunk 1]
```diff
@@ -17,6 +17,7 @@
 using Microsoft.ML.Internal.Utilities;
 using Microsoft.ML.Model;
 using Microsoft.ML.Trainers.FastTree;
+using Microsoft.ML.Transforms;
```
Comment (TomFinley) [inline src/Microsoft.ML.FastTree/GamModelParameters.cs]: >Transforms [](start = 19, length = 10)

I'm seeing a surprising number of these introductions. I don't object, but I am fairly curious what changed that they were no longer necessary in the past but are necessary now.

File: src/Microsoft.ML.KMeansClustering/KMeansModelParameters.cs [hunk 2]
```diff
@@ -13,13 +13,13 @@
 using Microsoft.ML.Model;
 using Microsoft.ML.Model.OnnxConverter;
 using Microsoft.ML.Numeric;
-using Microsoft.ML.Trainers.KMeans;
+using Microsoft.ML.Trainers;
 using Float = System.Single;
```
Comment (TomFinley) [inline src/Microsoft.ML.KMeansClustering/KMeansModelParameters.cs]: Ugh. One of you and @jwood803 are going to make each other very unhappy with the rebase conflicts. :D

File: src/Microsoft.ML.Transforms/doc.xml [hunk 3]
```diff
@@ -64,7 +64,7 @@
       <remarks>
         <para>
           This transform uses a set of aggregators to count the number of non-default values for each slot and
-          instantiates a <see cref="SlotsDroppingTransformer"/> to actually drop the slots.
+          instantiates a <see cref="T:Microsoft.ML.Transforms.SlotsDroppingTransformer"/> to actually drop the slots.
```
Comment (TomFinley) [inline src/Microsoft.ML.Transforms/doc.xml]: >Microsoft.ML.Transforms [](start = 38, length = 23)

Similar question, not sure why this suddenly became necessary when previously it was not. Or did we only just realize it was necessary? Totally fine, consistent with everythign else I see here, just a little odd.

File: src/Microsoft.ML.FastTree/GamModelParameters.cs (hunk 1 above)
Comment (sfilipi) [reply src/Microsoft.ML.FastTree/GamModelParameters.cs]: Good point! It is due to FeatureContributionCalculatingTransformer moving from Microsoft.ML.Data into Microsoft.ML.Transforms. 

Ml.Data was already there. 

---
In reply to: [260841129](https://github.com/dotnet/machinelearning/pull/2755#discussion_r260841129) [](ancestors = 260841129)

File: src/Microsoft.ML.Data/Transforms/FeatureContributionCalculationTransformer.cs [hunk 4]
```diff
@@ -22,7 +22,7 @@
 
 [assembly: LoadableClass(typeof(void), typeof(FeatureContributionEntryPoint), null, typeof(SignatureEntryPointModule), FeatureContributionCalculatingTransformer.LoaderSignature)]
 
-namespace Microsoft.ML.Data
+namespace Microsoft.ML.Transforms
```
Comment (sfilipi) [inline src/Microsoft.ML.Data/Transforms/FeatureContributionCalculationTransformer.cs]: Note, this is the only class that moved from ML.Data ->ML.Transforms. 

Everythign else was a sub-namespace of ML.Transforms or ML.Trainers

File: src/Microsoft.ML.Transforms/doc.xml (hunk 3 above)
Comment (sfilipi) [reply src/Microsoft.ML.Transforms/doc.xml]: this was not necessary; spotted the difference when changed the namespaces below, and i guess my OCD took over :)

---
In reply to: [260843071](https://github.com/dotnet/machinelearning/pull/2755#discussion_r260843071) [](ancestors = 260843071)

File: src/Microsoft.ML.Data/Transforms/FeatureContributionCalculationTransformer.cs (hunk 4 above)
Comment (TomFinley) [reply src/Microsoft.ML.Data/Transforms/FeatureContributionCalculationTransformer.cs]: Oh geez. That's good that you caught it!

File: src/Microsoft.ML.KMeansClustering/KMeansModelParameters.cs (hunk 2 above)
Comment (jwood803) [reply src/Microsoft.ML.KMeansClustering/KMeansModelParameters.cs]: I don't mind doing the merge on my pull request. :)

**Pull request merged**